tests/test_georaster.py

```diff
@@ -638,42 +681,43 @@ def test_value_at_coords(self) -> None:
         r2 = gr.Raster(self.landsat_b4_crop_path)
         r.crop(r2)
 
-        # random test point that raised an error
+        # Random test point that raised an error
         itest = 118
-        jtest = 516
-        xtest = 499540
-        ytest = 3099710
+        jtest = 450
+        xtest = 496930
+        ytest = 3099170
 
         # z = r.data[0, itest, jtest]
         x_out, y_out = r.ij2xy(itest, jtest, offset="ul")
         assert x_out == xtest
         assert y_out == ytest
 
-        # TODO: this fails, don't know why
-        # z_val = r.value_at_coords(xtest,ytest)
-        # assert z == z_val
+        z_val = r.value_at_coords(xtest, ytest)
+        z = r.data.data[0, itest, jtest]
+        assert z == z_val
 
-    def test_set_ndv(self) -> None:
+    @pytest.mark.parametrize('example', [landsat_b4_path, aster_dem_path])
+    def test_set_ndv(self, example: str) -> None:
         """
-        Read Landsat dataset and set 255 to no data. Save mask.
-        Then, set 254 as new no data (after setting 254 to 0). Save mask.
+        Read dataset and set a certain value (e.g., 255 or -9999) to no data. Save mask.
+        Then, set the value minus one as new no data (e.g., 254 or -10000), after rewriting the previous nodata to 0. Save mask.
         Check that both no data masks are identical and have correct number of pixels.
         """
-        # Read Landsat image and set no data to 255
-        r = gr.Raster(self.landsat_b4_path, masked=True)
+        # Read image and set no data to 255
+        r = gr.Raster(example, masked=True)
 
         # Copy the original data to validate the mask later
         original_data = r.data.copy()
-        # Set the nodata value to 255 and update the mask accordingly.
-        r.set_ndv(ndv=255, update_array=True)
+        # Set the nodata value to default ndv (e.g., 255) and update the mask accordingly.
+        r.set_ndv(ndv=_default_ndv(r.dtypes[0]), update_array=True)
         # Save the mask for validation
         ndv_index = r.data.mask.copy()
 
         # Now set to 254, after changing 254 to 0.
-        r.data[r.data == 254] = 0
+        r.data[r.data == _default_ndv(r.dtypes[0]) - 1] = 0
         # This will unset the mask of all masked 255 values
```

Comment:
See my comment regarding your issue #286. The tests show that this is the expected behavior.